molecule_vagrant/modules/vagrant.py

```diff
@@ -514,6 +514,9 @@ def _write_vagrantfile_config(self, data):
         )
 
     def _write_configs(self):
+        if self._module.params["state"] != "up":
```

Comment:
Wouldn't some operations fail if the configs aren't written? e.g. if a user runs `molecule destroy` before running `molecule create` then that would probably cause an error because the Vagrantfile wouldn't be written.

Reply:
I had the same fear but the tests were happy here so I guess it's fine ?

Reply:
I don't think we have tests for that specific situation, so they probably would pass :P Why not make `platform_box` mandatory in the destroy playbook?

Reply:
oh, yeah, we don't have test explicitely for this case. In fact, I should have written at least 3 tests for this commit:

1.  the one I wrote,
2. for testing that vm creation should fail if platform_box is specified
3. 1 for the case you're mentionning.

For 2, the playbooks are making sure of that. For 3, well the test sequence is:
```
    ├── cleanup
    ├── destroy
...
    ├── create
...
```
so it's implicitely tested.

As to make ``platform_box`` a requirement for destroy, I thought about it but:
- when using vagrant boxes, you're writing the ``Vagrantfile``, start the VM, destroy the VM, not write the ``Vagrantfile``, start the VM, write again the ``Vagrantfile``, destroy the VM
- it would make needless I/O
- it's probably the cause of bug 19
- it gives the feeling that multi nodes is working but it's more "by luck" than by purpose. While working on this, I now understand why my multinodes vagrant tests were acting weirdly in some cases.
- for configuration changes after VM creation, it's not really supported anyway I guess

Reply:
> so it's implicitely tested.

Ah, true.